libclamav/scanners.c

```diff
@@ -2082,6 +2082,69 @@ static cl_error_t cli_ole2_tempdir_scan_for_xlm_and_images(const char *dir, cli_
     return ret;
 }
 
+const char * const HTML_URLS_JSON_KEY = "HTMLUrls";
+
+
+
+static bool is_url(const char * const str){
+
+#define MATCH(str, prefix) \
+    do { \
+        if (str && (strlen(str) > strlen(prefix)) \
+                && (0 == strncasecmp(str, prefix, strlen(prefix)))) { \
+            bRet = true; \
+            goto done; \
+        } \
+    } while (0);
+
+    bool bRet = false;
+
+    MATCH(str, "https://");
```

Comment:
There's a canonical list of URIs [here](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml) and no reason I can see for limiting this to only 3 - if we're adding an option to extract URLs we should do it properly.

Reply:
I actually meant to go back and find a list once I had the code working, thank you!